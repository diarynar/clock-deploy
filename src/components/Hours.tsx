/** @jsxImportSource @emotion/react */

import { Divider } from './Divider'

interface HoursProps {
  hours: string[]
}
const styles = {
  container: {
    position: 'relative',
  },
}
function Hours({ hours }: HoursProps) {
  const hourClass = (hour: string) =>
    hour === 'R' ? 'bg-orange' : 'bg-grey-dark'

  return (
    <div data-testid="hours" className="flex flex-column">
      {hours.map((hour, i) => (
        <div key={i}>
          <div
            className="flex flex-row justify-center items-center w-18-rem bg-grey br-rounded"
            css={styles.container as any}
          >
            {hour &&
              hour
                .split('')
                .map((h, idx) => (
                  <div
                    key={idx}
                    className={`h-2-rem border br-rounded ${hourClass(h)}`}
                    css={{ width: 'calc(100% / 4)' }}
                  />
                ))}
          </div>
          <Divider />
        </div>
      ))}
    </div>
  )
}

export default Hours
