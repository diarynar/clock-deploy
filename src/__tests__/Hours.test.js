import { cleanup, render } from '@testing-library/react'

import  Hours  from '../components/Hours'

afterEach(cleanup)

describe('Hours should', () => {
  test('render 4 rectangle ', () => {
    const { container } = render(<Hours hours={['ROOO', 'ROOO']} />)
    expect(container.firstChild.firstChild).toMatchSnapshot()
  })
})
