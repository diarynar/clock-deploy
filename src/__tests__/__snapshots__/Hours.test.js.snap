// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hours should render 4 rectangle  1`] = `
<div>
  <div
    class="flex flex-row justify-center items-center w-18-rem bg-grey br-rounded css-w1m3xx-Hours"
  >
    <div
      class="h-2-rem border br-rounded bg-orange css-xhgmcw-Hours"
    />
    <div
      class="h-2-rem border br-rounded bg-grey-dark css-xhgmcw-Hours"
    />
    <div
      class="h-2-rem border br-rounded bg-grey-dark css-xhgmcw-Hours"
    />
    <div
      class="h-2-rem border br-rounded bg-grey-dark css-xhgmcw-Hours"
    />
  </div>
  <div
    class="flex flex-column items-center bg-grey css-1m94g40-Divider"
  >
    <p
      class="bg-dark css-17k9y5l-Divider"
    />
  </div>
</div>
`;
